ocw_data_parser/ocw_data_parser_test.py

```diff
@@ -2,29 +2,189 @@
+    parser = CustomHTMLParser(output_list=output_list)
+    assert "test" == parser.output_list[0]
+
+def test_parser_loaded_jsons(ocw_parser):
+    """
+    Test instantiating a parser with preloaded jsons
+    """
+    try:
+        assert OCWParser(loaded_jsons=ocw_parser.jsons)
+    except:
+        fail("instantiating parser with preloaded jsons failed")
+
+def test_generate_master_json_none_source(ocw_parser):
+    """
+    Make sure that running generate_master_json doesn't throw an error after nulling 
+    out the parser's source jsons
+    """
+    ocw_parser.jsons = None
+    ocw_parser.generate_master_json()
+    assert ocw_parser.jsons is not None
+
+def test_generate_master_json_none_course_image_uid(ocw_parser):
+    """
+    Make sure course_image_uid is regenerated by generate_master_json after it's nulled out
+    """
+    ocw_parser.course_image_uid = None
+    ocw_parser.generate_master_json()
+    assert ocw_parser.course_image_uid is not None
+
+def test_load_raw_jsons_invalid_file(ocw_parser):
+    """
+    Add a json file with invalid content to the course_dir and make sure it generates an error
+    """
+    with TemporaryDirectory() as destination_dir:
+        with open(os.path.join(constants.COURSE_DIR, "jsons/999.json"), "w") as f:
+            f.write("{")
+        try:
+            OCWParser(course_dir=constants.COURSE_DIR,
+                    destination_dir=destination_dir,
+                    static_prefix=constants.STATIC_PREFIX)
+            assert False
+        except:
+            assert True
+        os.remove(os.path.join(constants.COURSE_DIR, "jsons/999.json"))
+
+def test_upload_all_data_to_s3(ocw_parser_s3, s3_bucket):
+    """
+    Use moto (mock boto) to test s3 uploading
+    """
+    ocw_parser_s3.upload_all_media_to_s3(upload_master_json=True)
+    course_image_key = None
+    for bucket_item in s3_bucket.objects.filter(Prefix=ocw_parser_s3.s3_target_folder):
+        if bucket_item.key in ocw_parser_s3.course_image_s3_link:
+            course_image_key = bucket_item.key
+    assert course_image_key is not None
+
+def test_upload_course_image(ocw_parser_s3, s3_bucket):
+    """
+    Use moto (mock boto) to test s3 uploading
+    """
+    ocw_parser_s3.upload_course_image()
+    course_image_key = None
+    for bucket_item in s3_bucket.objects.filter(Prefix=ocw_parser_s3.s3_target_folder):
+        if bucket_item.key in ocw_parser_s3.course_image_s3_link:
+            course_image_key = bucket_item.key
+    assert course_image_key is not None
+
+def test_get_master_json(ocw_parser):
+    """
+    Test that getting the master JSON is not None or empty or missing basic properties 
+    and doesn't throw an exception
+    """
+    try:
+        master_json = ocw_parser.get_master_json()
+        assert master_json
+        assert master_json["uid"]
+        assert master_json["title"]
+        assert master_json["description"]
+        assert master_json["short_url"]
+    except:
```

Comment:
We should just let `assert` raise the exception so we can get the most detailed stack trace possible.